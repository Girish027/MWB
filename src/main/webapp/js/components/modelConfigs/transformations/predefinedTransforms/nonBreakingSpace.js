import Constants from 'constants/Constants';

const nonBreakingSpace = () => {
  const regex = {
    type: Constants.TRANSFORMATION_TYPES.REGEX_REPLACE,
    mappings: { },

  };
  const reg = /\xao/i;
  regex.mappings[reg] = ' ';

  return regex;
};

export default nonBreakingSpace;
