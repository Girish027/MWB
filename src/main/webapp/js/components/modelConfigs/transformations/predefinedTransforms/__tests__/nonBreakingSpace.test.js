import nonBreakingSpace from 'components/modelConfigs/transformations/predefinedTransforms/nonBreakingSpace';

describe('nonBreakingSpace', () => {
  const regex = nonBreakingSpace();

  test('type is equal to regex-replace', () => {
    expect(regex.type).toEqual('regex-replace');
  });

  test('mapping contains one key', () => {
    const keys = Object.keys(regex.mappings);
    expect(keys.length).toEqual(1);
  });

  test('snapshot predefined transformation', () => {
    expect(regex.mappings).toMatchSnapshot();
  });
});
