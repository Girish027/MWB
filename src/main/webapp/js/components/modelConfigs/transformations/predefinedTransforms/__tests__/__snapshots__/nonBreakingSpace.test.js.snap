// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nonBreakingSpace snapshot predefined transformation 1`] = `
Object {
  "/\\\\xao/i": " ",
}
`;
